core/shaders/fragment.js

```diff
@@ -0,0 +1,33 @@
+export const fragment = (defines) =>
```

Comment:
I'm not sure where to put `shaders` folder. Should it be in `materials`? Or `core` sounds better?

Reply:
I'd put it in the root, _next to_ `core` and `materials`. So far we managed to avoid putting nested directories in other folders.